web/client/components/TOC/fragments/settings/Display.jsx

```diff
@@ -261,7 +264,22 @@ export default class extends React.Component {
                         <Col xs={12} className={"legend-label"}>
                             <label key="legend-options-title" className="control-label"><Message msgId="layerProperties.legendOptions.title" /></label>
                         </Col>
-                        <Col xs={12} sm={6} className="first-selectize">
+                        { this.props.element?.serverType !== ServerTypes.NO_VENDOR &&
+                            <Col xs={12}>
+                                <Checkbox
+                                    data-qa="display-interactive-legend-option"
+                                    value="enableInteractiveLegend"
+                                    key="enableInteractiveLegend"
+                                    onChange={(e) => {
+                                        if (!e.target.checked) this.props.resetLegendFilter('disableEnableInteractiveLegend');
+                                        this.props.onChange("enableInteractiveLegend", e.target.checked);
```

Comment:
you can use something like this:

```
if (!e.target.checked) {
  const newLayerFilter = resetLayerLegendFilter(this.props.element.layerFilter);
  this.props.onChange("layerFilter", newLayerFilter );
}
```

where resetLayerLegendFilter is a utils that recieve a layerFilter object and return the updated layerFilter